### PR TITLE
chore(ci): CI with Github Actions

### PR DESCRIPTION
**Pull request by strophy**
## Issue being fixed or feature implemented
We experiencing several problems with Travis CI. It is slow and does not allow to restart builds.

## What was done?
Introduce Github Action workflows for running tests and releasing.

## How Has This Been Tested?
With Github Actions

## Breaking Changes
None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

/set-var install-sdk-version=3.18.1

### CODE REVIEW

**Comment by strophy** [comment]
Build output looks like this: https://www.npmjs.com/package/dash-strophy-test

File: .github/workflows/test_and_release.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      - uses: dashevo/gh-action-runtime-vars@master
+        id: runtime-vars
+        if: ${{ contains(github.event.pull_request.body, '/set-var')}}
+        with:
+          pull-request-body: ${{ github.event.pull_request.body }}
+
+      - uses: dashevo/gh-action-platform-branches@master
+        id: platform-branches
+        with:
+          override-dashmate-branch: ${{ steps.runtime-vars.outputs.dashmate-branch }}
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:55]: we need to override major version to `0`

File: .github/workflows/test_and_release.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        run: npm ci
+
+      - name: Create dotenv file
+        run: cp .env.example .env
+
+      - uses: dashevo/gh-action-runtime-vars@master
+        id: runtime-vars
+        if: ${{ contains(github.event.pull_request.body, '/set-var')}}
+        with:
+          pull-request-body: ${{ github.event.pull_request.body }}
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:50]: Why not just read it directly in action?

Comment (strophy) [reply .github/workflows/test_and_release.yml:50]: I opened a PR suggesting to do it this way 2 weeks ago: https://github.com/dashevo/gh-action-pr-vars/pull/1

File: .github/workflows/test_and_release.yml (hunk 1 above)
Comment (strophy) [reply .github/workflows/test_and_release.yml:56]: :+1: now I understand why this function exists

File: .github/workflows/test_and_release.yml (hunk 2 above)
Comment (shumkov) [reply .github/workflows/test_and_release.yml:47]: Merged! 👍

File: .github/workflows/test_and_release.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
+name: tests
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:1]: name: Test and Release

Could you please go through jobs and steps and make them canonical.

File: .github/workflows/test_and_release.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          faucet-private-key: ${{ env.faucet-private-key }}
+          dpns-contract-id: ${{ env.dpns-contract-id }}
+          dpns-top-level-identity-id: ${{ env.dpns-top-level-identity-id }}
+          dpns-top-level-identity-private-key: ${{ env.dpns-top-level-identity-private-key }}
+          platform-test-suite-branch: ${{ steps.platform-branches.outputs.testsuite-branch }}
+
+      - name: Show Docker logs
+        if: ${{ failure() }}
+        uses: jwalton/gh-docker-logs@v1
+
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:87]: We also need to run test suite

File: .github/workflows/test_and_release.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        run: |
+          echo "DAPI_SEED=127.0.0.1" > .env
+          echo "FAUCET_PRIVATE_KEY=${{ env.faucet-private-key }}" >> .env
+          echo "NETWORK=regtest" >> .env
+          echo "DPNS_CONTRACT_ID=${{ env.dpns-contract-id }}" >> .env
+
+      - name: Run SDK tests
+        run: npm run test
+
+      - uses: dashevo/gh-action-run-platform-test-suite@master
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:75]: Will it be called if previous step is failed?

Comment (strophy) [reply .github/workflows/test_and_release.yml]: No

File: .github/workflows/test_and_release.yml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+      - name: Start Dash local network
+        uses: dashevo/gh-action-start-local-network@master
+        with:
+          actions-cache-url: ${{ steps.platform-branches.outputs.cache-url }}
+          actions-cache-token: ${{ steps.platform-branches.outputs.cache-token }}
+          drive-branch: ${{ steps.platform-branches.outputs.current-branch }}
+          dapi-branch: ${{ steps.runtime-vars.outputs.dapi-branch }}
+          dashmate-branch: ${{ steps.platform-branches.outputs.dashmate-branch }}
+          install-sdk-version: ${{ steps.runtime-vars.outputs.install-sdk-version }}
```
Comment (shumkov) [inline .github/workflows/test_and_release.yml:64]: It suppose to be the an install path for `npm i` command as I remember. Something like `github:dashevo/js-dash-sdk#${{ current branch }}`.

**Review by shumkov — CHANGES_REQUESTED** [review]

File: .github/workflows/test_and_release.yml (hunk 6 above)
Comment (strophy) [reply .github/workflows/test_and_release.yml:64]: I'm still not sure what this option does or why it also works without the SDK. But it looks like the logic you are requesting here already exists in `gh-action-start-local-network`: https://github.com/dashevo/gh-action-start-local-network/blob/master/bin/start-local-node.sh#L56-L60

Comment (shumkov) [reply .github/workflows/test_and_release.yml:64]: It has SDK but we kinda overriding it with a specific version. This logic exist not just for dashmate, but for test suite as well, but slightly different https://github.com/dashevo/platform-test-suite/blob/v0.19-dev/bin/test.sh#L87-L90. So you need to pass current branch as `sdk-branch` to `start-local-network` and `github:dashevo/js-dash-sdk#${{ current branch }}` to  `run-platfrom-test-suite`.

Comment (strophy) [reply .github/workflows/test_and_release.yml:64]: I changed the way `gh-action-run-platform-test-suite` processes the variable [here instead](https://github.com/dashevo/gh-action-run-platform-test-suite/pull/1), so we don't need to make this change in all other repos using this action.

Comment (shumkov) [reply .github/workflows/test_and_release.yml:64]: Should we call then this input as `sdk-branch`?

File: .github/workflows/.env [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#DRIVE_BRANCH=
+DAPI_BRANCH=feat/get-status
```
Comment (shumkov) [inline .github/workflows/.env:2]: Let's keep it empty for now

**Comment by strophy** [comment]
Closed in favor of #204

**Pull request closed without merging**